nemo/core/classes/modelPT.py

```diff
@@ -91,6 +91,11 @@ def __init__(self, cfg: DictConfig, trainer: Trainer = None):
         # Convert config to support Hydra 1.0+ instantiation
         cfg = model_utils.maybe_update_config_version(cfg)
 
+        if 'model' in cfg:
+            raise ValueError(
+                "Creating model config node is forbidden due to collision problem when loading from checkpoint."
+            )
+
```

Comment:
Remove please.
What exactly is the collision?

Reply:
If this works correctly, please add a test that checks that conf.model passes this init but conf.model.model does not